network/transport/v2/protocol.go

```diff
@@ -71,22 +116,135 @@ func (p protocol) UnwrapMessage(envelope interface{}) interface{} {
 func (p protocol) Configure(_ transport.PeerID) {
 }
 
-func (p protocol) Start() {
+func (p *protocol) setupNatsHandler() error {
+	conn, err := p.natsConnectHandler(
+		p.config.Nats.Hostname,
+		p.config.Nats.Port,
+		time.Duration(p.config.Nats.Timeout)*time.Second,
+	)
+	if err != nil {
+		return err
+	}
+
+	js, err := conn.JetStream()
+	if err != nil {
+		return err
+	}
+
+	if _, err = js.Subscribe(events.PrivateTransactionsSubject, func(msg *nats.Msg) {
+		if err := p.handlePrivateTx(msg); err != nil {
+			logrus.Errorf("failed to handle private transaction: %v", err)
+
+			if err := msg.Nak(); err != nil {
+				logrus.Errorf("failed to NACK private transaction event: %v", err)
+			}
+
+			return
+		}
+
+		if err := msg.Ack(); err != nil {
+			logrus.Errorf("failed to ACK private transaction event: %v", err)
+		}
+	}); err != nil {
+		conn.Close()
+		return err
+	}
+
+	go func() {
+		<-p.ctx.Done()
+		conn.Close()
+	}()
+
+	return nil
+}
+
+func (p *protocol) Start() {
+	p.ctx, p.cancel = context.WithCancel(context.Background())
+
+	go func() {
+		for {
```

Comment:
Once disconnected, the subscription will not resume?

Reply:
Not sure what you mean. Note that the NATS client will automatically reconnect if the initial connection worked